packages/datatrak-web-server/src/utils/getFilteredUsers.ts

```diff
@@ -73,3 +53,37 @@ export const getFilteredUsers = async (
     }))
     .slice(0, DEFAULT_PAGE_SIZE);
 };
+
+export const getFilteredUsersForPermissionGroup = async (
+  models: DatatrakWebServerModelRegistry,
+  countryCode: Country['code'],
+  permissionGroup: PermissionGroupRecord,
+  searchTerm?: string,
+) => {
+  // if the permission group is a public permission group that every user has access to because of the api client permissions, then everyone has access to the survey, so return all non-internal users
+  if (
+    API_CLIENT_PERMISSIONS.find(
+      ({ entityCode, permissionGroupName }) =>
+        entityCode === countryCode && permissionGroupName === permissionGroup.name,
+    )
+  ) {
+    return getFilteredUsers(models, searchTerm);
+  }
+
+  // get the ancestors of the permission group
+  const permissionGroupWithAncestors = await permissionGroup.getAncestors();
+  const entity = await models.entity.findOne({
+    country_code: countryCode,
+    type: EntityTypeEnum.country,
+  });
```

Comment:
Bit of over-`await`ing here (even though it is easier to read 🥲)

```suggestion
  const [permissionGroupWithAncestors, entity] = await Promise.all([
		permissionGroup.getAncestors(),
		models.entity.findOne({
			country_code: countryCode,
			type: EntityTypeEnum.country,
		}),
	]);
```